Makefile

```diff
@@ -310,7 +310,8 @@ test-single-suite: ## Run a single test `make test-single-suite TEST_NAME=<test
 			if [[ '${BREAK_ON_FAILURE__}' = 'TRUE' ]];then echo "Breaking on first failure! (iteration $$i)"; exit $${RESULT}; fi; fi; \
 	done; \
 	echo "Test ${TEST_NAME} completed ${NUM_REPEATS__} iterations" \
-		"($$((${NUM_REPEATS__}-num_failures)) succeed, $${num_failures} failed)";
+		"($$((${NUM_REPEATS__}-num_failures)) succeed, $${num_failures} failed)"; \
+	exit $${num_failures}
```

Comment:
AFAIK, the issue is only seen with the Restart Recovery job.
On the other hand, it looks like this fix would affect all Apollo jobs that run as part of CI. Why aren't we seeing this issue with other jobs then?

Reply:
I've checked this and the return code which script returns by default is of the last executed command which in this case is "echo ...". Following script returns 0. If u comment last "ls" it will return code 2.
`#! /bin/bash
ls a_file_which_doesnt_exists
ls
`

Reply:
I think retries are expected in some cases (for flaky tests for instance).
With your fix, in case of retries, the suite will be considered failed, which we don't want.